### PR TITLE
chore (updatecli): run updatecli with jenkins

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3419
this depend on the PR adding the folder in kubernet management on infra.ci like : https://github.com/jenkins-infra/kubernetes-management/blob/1ee4ea721c8285c7bd28e3518d835f0e3b045ca2/config/ext_jenkins-infra-jobs.yaml#L152

should be completed by https://github.com/jenkins-infra/kubernetes-management/pull/3648

### CODE REVIEW

**Comment by smerle33** [comment]
in fact this PR will be merged after the https://github.com/jenkins-infra/kubernetes-management/pull/3648 in order to check the pipeline

**Review by MarkEWaite — DISMISSED** [review]
Ugh, I used my command line approval script on the wrong pull request.  Apologies for the unintended approval

**Review by lemeurherve — DISMISSED** [review]

File: Jenkinsfile_updatecli [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      }
+    } // stage
+    stage('Apply Configuration Update') {
+      when {
+        expression { env.BRANCH_IS_PRIMARY }
+      }
+      steps {
+        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+          // updatecli(action: 'apply',  config: './updatecli/weekly.d', cronTriggerExpression: cronExpr)
+          sh 'updatecli version'
```
Comment (lemeurherve) [inline Jenkinsfile_updatecli]:
```suggestion
```
Already in the previous stage

**Review by lemeurherve — COMMENTED** [review]
The files https://github.com/jenkins-infra/jenkins-infra/blob/production/.github/workflows/updatecli.yaml and https://github.com/jenkins-infra/jenkins-infra/blob/production/updatecli/values.github-action.yaml should be deleted in this PR too.

**Comment by lemeurherve** [comment]
I'm curious, why calling updatecli directly instead of using the `updatecli` shared pipeline library function?

**Comment by smerle33** [comment]
> The files https://github.com/jenkins-infra/jenkins-infra/blob/production/.github/workflows/updatecli.yaml and https://github.com/jenkins-infra/jenkins-infra/blob/production/updatecli/values.github-action.yaml should be deleted in this PR too.

I plan it on another PR after this one is merged and the builds are ok on the production branch

**Comment by smerle33** [comment]
> I'm curious, why calling updatecli directly instead of using the `updatecli` shared pipeline library function?

because we need a `docker` command ... so using a VM ... and the current version of the shared pipeline library imply using a pod ... so no docker inside.

File: Jenkinsfile_updatecli [hunk 2]
```diff
@@ -0,0 +1,58 @@
+def cronExpr = env.BRANCH_IS_PRIMARY ? 'H/30 * * * *' : ''
+
+pipeline {
+  agent { label 'linux-amd64-docker' }
```
Comment (dduportal) [inline Jenkinsfile_updatecli:5]:
```suggestion
  // `docker` and `updatecli` are required
  agent { label 'linux-amd64-docker' }
```

nitpick: adding a comment to write down the constraint

File: Jenkinsfile_updatecli [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          }
+        }
+      }
+    } // stage
+    stage('Apply Configuration Update') {
+      when {
+        expression { env.BRANCH_IS_PRIMARY }
+      }
+      steps {
+        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
```
Comment (dduportal) [inline Jenkinsfile_updatecli]:
```suggestion
```

Blocking: if the "updatecli apply" step fails, then the pipeline must fail. The `catchError()` is only for `updatecli diff` to handle cases where a new code needs to be merged on the principal branch before updatecli can handle it.

File: Jenkinsfile_updatecli [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          withCredentials([
+            usernamePassword(
+            credentialsId: 'github-app-updatecli-on-jenkins-infra',
+            usernameVariable: 'USERNAME_VALUE', // Setting this variable is mandatory, even if of not used when the credentials is a githubApp one
+            passwordVariable: 'UPDATECLI_GITHUB_TOKEN'
+            )
+          ]) {
+            sh 'updatecli apply --config ./updatecli/weekly.d --values ./updatecli/values.yaml'
+          }
+        }
```
Comment (dduportal) [inline Jenkinsfile_updatecli]:
```suggestion
```

this change is coupled with removing the `cathcError` for apply

**Review by dduportal — CHANGES_REQUESTED** [review]
Nice job. LGTM, except that you have to remove the `catchError` for the apply part.

**Review by dduportal — APPROVED** [review]
LGTM 👍 Let's merge and see how it behaves

**Pull request merged**